api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -51,26 +53,42 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	public Cohort getCohort(Integer id) throws DAOException {
 		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(java.lang.Integer)
+	 */
+	@SuppressWarnings("unchecked")
+	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean voided) throws DAOException {
+		Disjunction orDisjunction = Restrictions.disjunction();
+		orDisjunction.add(Restrictions.isNull("m.endDate"));
+		orDisjunction.add(Restrictions.gt("m.endDate", new Date()));
+		return (List<Cohort>) sessionFactory.getCurrentSession().createCriteria(Cohort.class)
+				.add(Restrictions.eq("voided", voided))
+				.createAlias("members", "m")
+				.add(Restrictions.le("m.startDate", new Date()))
+				.add(orDisjunction)
+				.createAlias("m.patient", "p")
+				.add(Restrictions.eq("p.patientId", patientId))
```

Comment:
You don't need this restriction and createAlias calls, I think you should have one restriction in place of the 2 lines above like this:

`Restrictions.eq("m.patient", new Patient(1))`

Alternatively you can load the patient by the patientId first and use the loaded patient object in place of of `new Patient(1)`